From f69edd2042ebf1db72ad1a2c6cbfd8887231da25 Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Wed, 7 Jun 2023 15:15:42 +0200
Subject: [PATCH] =?UTF-8?q?shared:=20move=20uid-alloc-range.[ch]=20from=20?=
 =?UTF-8?q?src/shared/=20=E2=86=92=20src/basic/?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This way we can use it in libsystemd

(cherry picked from commit 8a8b4a8784f48e941b6b460b4cb963929d1d6a8d)

Related: RHEL-31070
---
 src/basic/meson.build                   | 2 ++
 src/{shared => basic}/uid-alloc-range.c | 0
 src/{shared => basic}/uid-alloc-range.h | 0
 src/shared/meson.build                  | 2 --
 4 files changed, 2 insertions(+), 2 deletions(-)
 rename src/{shared => basic}/uid-alloc-range.c (100%)
 rename src/{shared => basic}/uid-alloc-range.h (100%)

diff --git a/src/basic/meson.build b/src/basic/meson.build
index bfe52d5879..c0f0b07418 100644
--- a/src/basic/meson.build
+++ b/src/basic/meson.build
@@ -240,6 +240,8 @@ basic_sources = files(
         'time-util.h',
         'tmpfile-util.c',
         'tmpfile-util.h',
+        'uid-alloc-range.c',
+        'uid-alloc-range.h',
         'uid-range.c',
         'uid-range.h',
         'umask-util.h',
diff --git a/src/shared/uid-alloc-range.c b/src/basic/uid-alloc-range.c
similarity index 100%
rename from src/shared/uid-alloc-range.c
rename to src/basic/uid-alloc-range.c
diff --git a/src/shared/uid-alloc-range.h b/src/basic/uid-alloc-range.h
similarity index 100%
rename from src/shared/uid-alloc-range.h
rename to src/basic/uid-alloc-range.h
diff --git a/src/shared/meson.build b/src/shared/meson.build
index 9e11e13934..766e4f9506 100644
--- a/src/shared/meson.build
+++ b/src/shared/meson.build
@@ -311,8 +311,6 @@ shared_sources = files(
         'tpm2-util.h',
         'udev-util.c',
         'udev-util.h',
-        'uid-alloc-range.c',
-        'uid-alloc-range.h',
         'user-record-nss.c',
         'user-record-nss.h',
         'user-record-show.c',
